HunterPie.Core/Game/World/MHWGame.cs

```diff
@@ -125,25 +126,36 @@ private void GetTimeElapsed()
                 if (Player.InHuntingZone)
                 {
                     // No quest timer available while player is on the hunt.
-                    // We will use ours instead.
-                    _localTimerStopwatch.Start();
-                    TimeElapsed = _localTimerStopwatch.ElapsedMilliseconds / 1000.0f;
-                }
-                else
+                    // We will use our local timer instead.
+                    if (_localTimerStopwatch.IsRunning)
+                    {
+                        this.SetTimeElapsed(_localTimerStopwatch.ElapsedMilliseconds / 1000.0f, false);
+                    } else
+                    {
+                        _localTimerStopwatch.Start();
+                        this.SetTimeElapsed(0, true);
+                    }
+                } else
                 {
                     // Prevent TimeElapsed from being 3000 sec. before joining the hunt.
                     // Otherwise there will be incorrect MemberInfo.JoinedAt values when player is entering Training Area or Guiding Lands.
-                    TimeElapsed = 0;
+                    this.SetTimeElapsed(0, false);
                 }
-            }
-            else
+            } else
             {
-                if (_localTimerStopwatch.IsRunning)
-                    _localTimerStopwatch.Reset();
                 float questMaxTimer = questMaxTimerRaw
                     .ApproximateHigh(MHWGameUtils.MaxQuestTimers)
                     .ToSeconds();
-                TimeElapsed = Math.Max(0, questMaxTimer - elapsed);
+                var timeElapsed = Math.Max(0, questMaxTimer - elapsed);
+                if (_localTimerStopwatch.IsRunning)
+                {
+                    // Stop local timer before switching to the real quest timer.
+                    _localTimerStopwatch.Reset();
+                    this.SetTimeElapsed(timeElapsed, true);
+                } else
+                {
+                    this.SetTimeElapsed(timeElapsed, false);
```

Comment:
Please keep the code style like the rest of the project, `this` should not be used. The only exception is in the `this.Dispatch` because you can't access that extension otherwise.

Reply:
Actually I was thinking there could be something wrong with my auto formatting config. When I was typing
```c#
}
else
{
```
I got this
```c#
} else
{
```

After rebasing the formatting behaves as expected, though.